apps/src/templates/progress/BubbleFactory.jsx

```diff
@@ -137,6 +173,22 @@ export function getBubbleClassNames(isEnabled) {
   return classNames('progress-bubble', {enabled: isEnabled});
 }
 
+export function getBubbleUrl(levelUrl, studentId, sectionId) {
```

Comment:
@clareconstantine FYI, here's another instance that could benefit from the url parsing work you're doing.